### PR TITLE
Update to Kotlin 1.8

### LINKED ISSUE #119: With Kotlin 1.8, kapt tries to compile generated Java classes using Kotlin
When updating our project from Kotlin 1.7.10 to 1.8.10, I got an error for all the `@GenSealedEnum` companion objects in our project. For some reason `kapt` was expecting Kotlin classes, but the generated classes were Java code. Here's an example of the error message:

```
[project]/app/build/tmp/kapt3/stubs/debug/com/[redacted]/processor/pricelist/PriceSortOrder.java:139: error: Annotated element is not a Kotlin class
    public static final class Companion {
```
I was able to switch to `ksp` and everything generated properly.

Here's relevant lines from `build.gradle.kts`

```
buildscript {
   dependencies {
       ...
       classpath(kotlin("gradle-plugin", version = "1.7.10"))
       ...
   }
}

...

plugins {
    id("com.android.application")
    kotlin("android")
    kotlin("kapt")
    ...
}

dependencies {
    ...
    implementation("com.github.livefront.sealed-enum:runtime:0.5.0")
    kapt("com.github.livefront.sealed-enum:processor:0.5.0")
    ...
}
```

### PR DESCRIPTION
**Pull request by alexvanyo**
Updates the project to Kotlin 1.8, along with a few other dependencies.

- [x] Unmark as draft once kotlinpoet has a release of `1.13.0` with Kotlin 1.8 updates, currently using snapshot

Fixes #119

### CODE REVIEW

File: config/detekt/detekt.yml
Comment (alexvanyo) [inline config/detekt/detekt.yml:1]: I removed most of `detekt.yml` to make better use of the `buildUponDefaultConfig`

Before doing that, the version bump to `detekt` resulted in some various deprecation warnings that occurred in some rules.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 incap = "0.3"
-jacoco = "0.8.7"
+jacoco = "0.8.8"
 junit = "5.8.1"
-kotlin = "1.7.0"
-kotlinxBinaryCompatibilityValidator = "0.11.0"
-kotlinCompileTesting = "1.4.9"
-kotlinPoet = "1.12.0"
-ksp = "1.7.0-1.0.6"
+kotlin = "1.8.0"
```
Comment (brian-livefront) [inline gradle/libs.versions.toml:8]: 👍

File: ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SealedEnumProcessor.kt [hunk 2]
```diff
@@ -98,7 +98,6 @@ internal class SealedEnumProcessor(
         resolver: Resolver,
         sealedClassCompanionObjectKSClass: KSClassDeclaration
     ): SealedEnumFileSpec? {
-
```
Comment (brian-livefront) [inline ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SealedEnumProcessor.kt:101]: Was this required by the change in detekt?

File: ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SuperInterfaces.kt [hunk 3]
```diff
@@ -65,9 +65,9 @@ internal class SuperInterfaces(
             .filter {
                 val declaration = it.declaration
                 declaration is KSClassDeclaration &&
-                        declaration.classKind == ClassKind.INTERFACE &&
-                        declaration.isVisibleInterface() &&
-                        !declaration.isSealedInterface()
+                    declaration.classKind == ClassKind.INTERFACE &&
```
Comment (brian-livefront) [inline ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SuperInterfaces.kt:68]: Are these changes all required by detekt now?

File: processing-common/src/main/kotlin/com/livefront/sealedenum/internal/common/spec/SealedEnumOrdinalPropertySpec.kt [hunk 4]
```diff
@@ -8,7 +8,6 @@ import com.squareup.kotlinpoet.TypeName
 import javax.lang.model.element.TypeElement
 
 internal data class SealedEnumOrdinalPropertySpec(
-    private val sealedClass: SealedClass,
```
Comment (brian-livefront) [inline processing-common/src/main/kotlin/com/livefront/sealedenum/internal/common/spec/SealedEnumOrdinalPropertySpec.kt:11]: Was this just unused then?

Comment (alexvanyo) [reply processing-common/src/main/kotlin/com/livefront/sealedenum/internal/common/spec/SealedEnumOrdinalPropertySpec.kt:11]: Yep, the update to detekt picked up the unused variable, so I just removed it.

File: ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SealedEnumProcessor.kt (hunk 2 above)
Comment (alexvanyo) [reply ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SealedEnumProcessor.kt:101]: Yes, this change was enforced and auto-corrected by detekt.

File: ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SuperInterfaces.kt (hunk 3 above)
Comment (alexvanyo) [reply ksp/src/main/kotlin/com/livefront/sealedenum/internal/ksp/SuperInterfaces.kt:68]: Yeah, it seems like it changed the indention logic a touch

**Review by brian-livefront — APPROVED** [review]
Looks good!

**Comment by alexvanyo** [comment]
Rebased on top of `main` for new set of checks to take effect.

**Review by brian-livefront — APPROVED** [review]

**Pull request merged**